rest-openehr/src/main/java/org/ehrbase/rest/openehr/OpenehrQueryController.java

```diff
@@ -249,28 +248,45 @@ private void createRestContext(String qualifiedName, @Nullable String version) {
                 QUERY_EXECUTE_ENDPOINT,
                 Boolean.TRUE,
                 HttpRestContext.LOCATION,
-                fromPath("").pathSegment(QUERY, qualifiedName, version).build().toString());
+                fromPath("")
+                        .pathSegment(Q_PARAM, qualifiedName, version)
+                        .build()
+                        .toString());
     }
 
     @SuppressWarnings("unchecked")
-    private QueryResultDto invoke(@NonNull String queryString, Map<String, Object> requestBody) {
+    private AqlQueryRequest createRequest(@NonNull String queryString, Map<String, Object> requestBody) {
 
         requestBody = Optional.ofNullable(requestBody).orElseGet(Map::of);
         Map<String, Object> queryParameters = Optional.ofNullable(requestBody.get(QUERY_PARAMETERS))
                 .map(p -> (Map<String, Object>) p)
                 .orElseGet(Map::of);
-        Optional<Long> fetch = optionalLong(FETCH, requestBody);
-        Optional<Long> offset = optionalLong(OFFSET, requestBody);
+        Optional<Long> fetch = optionalLong(FETCH_PARAM, requestBody);
+        Optional<Long> offset = optionalLong(OFFSET_PARAM, requestBody);
 
-        return invoke(queryString, queryParameters, fetch, offset);
+        return createRequest(queryString, queryParameters, fetch, offset);
     }
 
-    private QueryResultDto invoke(
+    private AqlQueryRequest createRequest(
             @NonNull String queryString, Map<String, Object> parameters, Optional<Long> fetch, Optional<Long> offset) {
 
-        AqlQueryRequest aqlQueryRequest =
-                new AqlQueryRequest(queryString, parameters, fetch.orElse(null), offset.orElse(null));
-        return aqlQueryService.query(aqlQueryRequest);
+        return new AqlQueryRequest(queryString, parameters, fetch.orElse(null), offset.orElse(null));
+    }
+
+    protected QueryResponseData createQueryResponse(
+            QueryResultDto aqlQueryResult, String queryString, @Nullable URI location) {
+        final QueryResponseData queryResponseData = new QueryResponseData(aqlQueryResult);
+        queryResponseData.setQuery(queryString);
```

Comment:
can be removed because it's assigned here https://github.com/ehrbase/openEHR_SDK/blob/develop/response-dto/src/main/java/org/ehrbase/openehr/sdk/response/dto/QueryResponseData.java#L59